projects/igniteui-angular/src/lib/action-strip/grid-actions/grid-actions-base.directive.ts

```diff
@@ -0,0 +1,17 @@
+import { Directive, Input } from '@angular/core';
+
+@Directive({
+    selector: '[igxGridActionsBase]'
+})
+export class IgxGridActionsBaseDirective {
+    /**
+     * An @Input property that set an instance of the grid for which to display the actions.
+     * ```html
+     *  <igx-grid-pinning-actions [grid]="grid1"></igx-grid-pinning-actions>
+     *  <igx-grid-editing-actions [grid]="grid1"></igx-grid-editing-actions>
+     * ```
+     */
+    @Input() grid;
+
+    @Input() context;
```

Comment:
Assuming we will have a service that will pass the context, and the context for grid actions can be either row or cell, both of which have reference to the grid they belong to, these Inputs would probably not be necessary. Both can be getters that read from the service.

That will also make it easier to implement in a hierarchical grid scenario where you will not have a direct reference for child grid in your template that you can pass to the grid action `grid` input. It makes more sense for the action strip service to extract the grid from the cell/row context, rather than to require the user to explicitly set it.